extensions/wikia/Email/Controller/WatchedPageController.class.php

```diff
@@ -26,8 +33,11 @@ public function initEmail() {
 
 		$this->title = \Title::newFromText( $titleText, $titleNamespace );
 		$this->summary = $this->getVal( 'summary' );
-		$this->currentRevId = $this->getVal( 'currentRevId' );
-		$this->previousRevId = $this->getVal( 'previousRevId' );
+
+		if ( $this->title instanceof \Title ) {
+			$this->currentRevId = $this->title->getLatestRevID( \Title::GAID_FOR_UPDATE );
+			$this->previousRevId = $this->title->getPreviousRevisionID( $this->currentRevId, \Title::GAID_FOR_UPDATE );
+		}
```

Comment:
So does it not send any value for the ```*RevId``` or does it send the wrong value?  If it doesn't send any value, could we maybe check for that and only look it up if it wasn't passed?  Otherwise if they are sent but they're wrong, maybe we could check whether we're handling a delete/move/protect and only look it up then.  My main concern is just the race condition issue where we'd create a diff URL For a different change than the one made.